pretix_sepadebit/tests/test_payment.py

```diff
@@ -0,0 +1,279 @@
+        (now() + timedelta(days=14), 7, now() + timedelta(days=7), now().date() + timedelta(days=14)),
+        (now() + timedelta(days=14), 7, now() + timedelta(days=8), now().date() + timedelta(days=15)),
+        (now() + timedelta(days=14), 7, now() + timedelta(days=9), now().date() + timedelta(days=16)),
+        # Only prenotification based _due_date computation
+        (None, 7, now() - timedelta(days=123), now().date() + timedelta(days=7) - timedelta(days=123)),
+        (None, 7, now(), now().date() + timedelta(days=7)),
+        (None, 7, now() + timedelta(days=1), now().date() + timedelta(days=7) + timedelta(days=1)),
+        (None, 7, now() + timedelta(days=2), now().date() + timedelta(days=7) + timedelta(days=2)),
+        (None, 7, now() + timedelta(days=3), now().date() + timedelta(days=7) + timedelta(days=3)),
+        (None, 7, now() + timedelta(days=4), now().date() + timedelta(days=7) + timedelta(days=4))
+    ]
+)
+@pytest.mark.django_db
+def test_due_date_with_earliest_due_date(event, earliest_due_date, prenotification_days, order_date, due_date):
+    if earliest_due_date:
+        event.settings.set('payment_sepadebit_earliest_due_date', earliest_due_date)
+    event.settings.set('payment_sepadebit_prenotification_days', prenotification_days)
+    sepa = SepaDebit(event)
+
+    o = Order.objects.create(
+        code='123AB', event=event,
+        status=Order.STATUS_PENDING,
+        datetime=order_date, expires=order_date + timedelta(days=10),
+        total=23,
+    )
+
+    assert due_date == sepa._due_date(o)
+
+
+@pytest.mark.django_db
+def test_create_sepaduedate_instances(event, order):
+    with scopes_disabled():
+        op_date = now().date()
+        remind_after=now()+timedelta(days=1)
+        op = orderpayment(order, op_date, remind_after=remind_after)
+
+        migration.create_sepaduedate_instances(OrderPayment, SepaDueDate)
+
+        op.refresh_from_db()
+
+        assert op.due.date == op_date
+        with pytest.raises(KeyError):
+            op.info_data['date']
+        assert op.info_data['testdata'] == 'is not deleted'
+        assert op.due.reminded == True
+        assert op.due.remind_after.date() == op.due.date
+
+
+@pytest.mark.django_db
+def test_create_sepaduedate_no_instances(event):
+    with scopes_disabled():
+        try:
+            migration.create_sepaduedate_instances(OrderPayment, SepaDueDate)
+        except Exception:
+            pytest.fail("Shouldn't raise exception if no matching OrderPayment exists.")
+
+
+@pytest.mark.django_db
+def test_delete_sepaduedate_instances(event, order):
+    with scopes_disabled():
+        op_date = now().date()
+        remind_after=now()+timedelta(days=1)
+        op = orderpayment(order, date=op_date, reminded=True, remind_after=remind_after, old_format=False)
+
+        migration.delete_sepaduedate_instances(OrderPayment, SepaDueDate)
```

Comment:
This is the first PR I've ever seen including tests for a migration. And even for the backwards path. I love the attention to detail <3

Reply:
Funny what being somewhat scared can lead up to 😅
This was the original reason for the non-standard migration name. Had to do some digging to avoid it